From 4602261fa44d6bbb4c3698c79e08a6a40a6edc5a Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 12 Jan 2024 09:21:39 -0800
Subject: [PATCH] clang/arm: Do not use MUST_TAIL_CALL

This causes clang-17 to crash see [1]
this code is new in webkit 2.42[2] thats why we do not see the crash in older webkit

[1] https://github.com/llvm/llvm-project/issues/67767
[2] https://github.com/WebKit/WebKit/commit/4d816460b765acd8aef90ab474615850b91ecc35

Upstream-Status: Inappropriate [work around to avoid clang compiler crash]
Signed-off-by: Khem Raj <raj.khem@gmail.com>

Update context for webkitgtk 2.48.0.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 Source/WTF/wtf/Compiler.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WTF/wtf/Compiler.h b/Source/WTF/wtf/Compiler.h
index 16e416d..68dd9a0 100644
--- a/Source/WTF/wtf/Compiler.h
+++ b/Source/WTF/wtf/Compiler.h
@@ -293,7 +293,7 @@
 #if COMPILER(CLANG)
 #if __SIZEOF_POINTER__ == 8
 #if !defined(MUST_TAIL_CALL) && defined(__cplusplus) && defined(__has_cpp_attribute)
-#if __has_cpp_attribute(clang::musttail) && !defined(__powerpc__) && !defined(_WIN32)
+#if __has_cpp_attribute(clang::musttail) && !defined(__powerpc__) && !defined(_WIN32) && !defined(__arm__)
 #define MUST_TAIL_CALL [[clang::musttail]]
 #define HAVE_MUST_TAIL_CALL 1
 #endif
